src/appleseed/foundation/platform/python.h

```diff
@@ -42,6 +42,14 @@
 #include "boost/python/detail/wrap_python.hpp"  // has to be first, to avoid redefinition warnings
 #include "boost/python.hpp"
 
+// wrap_python.hpp undefines _DEBUG and then redefines it without assigning it value of 1
+// In OSL 1.11.8 osl/platform.h header uses the syntax "#if _DEBUG" which assumes
+// that _DEBUG is not an empty macro, otherwise the header can't be compiled
+#ifdef _DEBUG
+  #undef _DEBUG
+  #define _DEBUG 1
+#endif
```

Comment:
So there is a conflit between boost and OSL ?

Why putting this here exactly ? There isn't any OSL include bellow those lines.

Reply:
Yes. `wrap_python.hpp` undefines "_DEBUG" then redefines it as an empty macro.
https://github.com/boostorg/python/blob/f1320bf30b4887aff28b9fc67b98f6f8be8c3684/include/boost/python/detail/wrap_python.hpp#L45
https://github.com/boostorg/python/blob/f1320bf30b4887aff28b9fc67b98f6f8be8c3684/include/boost/python/detail/wrap_python.hpp#L209

I added this here directly after "wrap_python.hpp" to fix the macro definition.
Personally I think that the syntax `#if OSL_DEBUG` instead of `#ifdef OSL_DEBUG` used in OSL is suboptimal.

Reply:
I see. We have some headers in appleseed wich goal is to include another header and then fix some things, like you are doing here. This ensure that the fix will be avilable everywhere we need `wrap_python` without having to write the fix multiple times.

@dictoon Can you confirm this ? Do you have an example in mind ?